tokio-trace-fmt/src/default.rs

```diff
@@ -10,44 +14,163 @@ use std::fmt::{self, Write};
+    }
+}
+#[cfg(not(feature = "chrono"))]
+impl FormatTime for SystemTime {
+    fn format_time(&self, w: &mut fmt::Write) -> fmt::Result {
+        write!(w, "{:?} ", std::time::SystemTime::now())
+    }
+}
+
+/// Builder for a `Standard` formatter.
+#[derive(Default, Debug, Clone)]
+pub struct Builder<T = SystemTime> {
+    full: bool,
+    timer: T,
+}
+
+impl<T> Builder<T> {
+    /// Use a more verbose output format.
+    pub fn full(mut self) -> Self {
+        self.full = true;
+        self
+    }
+
+    /// Use the given `timer` for log message timestamps.
+    pub fn with_timer<T2>(self, timer: T2) -> Builder<T2>
+    where
+        T2: FormatTime,
+    {
+        Builder {
+            full: self.full,
+            timer,
+        }
+    }
+
+    /// Do not emit timestamps with log messages.
+    pub fn without_time(self) -> Builder<()> {
+        Builder {
+            full: self.full,
+            timer: (),
+        }
+    }
+
+    /// Produce a `Standard` event formatter from this `Builder`'s configuration.
+    pub fn build(self) -> Standard<T> {
+        Standard {
+            full: self.full,
+            timer: self.timer,
+        }
+    }
+}
+
+/// A pre-configured event formatter.
+///
+/// You will usually want to use this as the `Formatter` for a `FmtSubscriber`.
+#[derive(Debug, Clone)]
+pub struct Standard<T = SystemTime> {
+    full: bool,
+    timer: T,
+}
+
+impl<T> Default for Standard<T>
 where
-    N: for<'a> ::NewVisitor<'a>,
+    T: Default,
 {
-    let meta = event.metadata();
-    write!(
-        f,
-        "{} {}{}: ",
-        FmtLevel(meta.level()),
-        FmtCtx(&ctx),
-        meta.target()
-    )?;
-    {
-        let mut recorder = ctx.new_visitor(f, true);
-        event.record(&mut recorder);
+    fn default() -> Self {
+        Builder::default().build()
     }
-    ctx.with_current(|(_, span)| write!(f, " {}", span.fields()))
-        .unwrap_or(Ok(()))?;
-    writeln!(f)
 }
 
-pub fn fmt_verbose<N>(ctx: &span::Context<N>, f: &mut Write, event: &Event) -> fmt::Result
+impl<N, T> Formatter<N> for Standard<T>
 where
     N: for<'a> ::NewVisitor<'a>,
+    T: FormatTime,
 {
-    let meta = event.metadata();
-    write!(
-        f,
-        "{} {}{}: ",
-        FmtLevel(meta.level()),
-        FullCtx(&ctx),
-        meta.target()
-    )?;
-    {
-        let mut recorder = ctx.new_visitor(f, true);
-        event.record(&mut recorder);
+    fn format(
+        &self,
+        ctx: &span::Context<N>,
+        writer: &mut fmt::Write,
+        event: &Event,
+    ) -> fmt::Result {
+        let meta = event.metadata();
+        {
+            #[cfg(feature = "ansi")]
+            let style = Style::new().dimmed();
+            #[cfg(feature = "ansi")]
+            write!(writer, "{}", style.prefix())?;
+            self.timer.format_time(writer)?;
+            #[cfg(feature = "ansi")]
+            write!(writer, "{}", style.suffix())?;
+        }
+        if self.full {
```

Comment:
It would be nice if this was dispatched based on types, rather than a bool (as it is in the previous approach). Not a blocker, we can change this later.